pkg/upstream/http.go

```diff
@@ -100,17 +101,37 @@ func newReverseProxy(target *url.URL, upstream options.Upstream, errorHandler Pr
 	if upstream.FlushInterval != nil {
 		proxy.FlushInterval = *upstream.FlushInterval
 	} else {
-		proxy.FlushInterval = 1 * time.Second
+		proxy.FlushInterval = options.DefaultFlushInterval
 	}
 
+	var finalTimeout time.Duration
+	if upstream.Timeout != nil {
+		finalTimeout = *upstream.Timeout
+	} else {
+		finalTimeout = options.DefaultTimeout
+	}
+
+	var finalKeepAlive time.Duration
+	if upstream.Timeout != nil {
+		finalKeepAlive = *upstream.KeepAlive
+	} else {
+		finalKeepAlive = options.DefaultKeepAlive
+	}
+
+	defaultTransportClone := http.DefaultTransport.(*http.Transport).Clone()
+	defaultTransportClone.DialContext = (&net.Dialer{
+		Timeout:   finalTimeout,
```

Comment:
So I wonder if looking under the hood, if `http.Transport.IdleConnectTimeout` is actually the thing we want to tinker with to prevent upstream connections from disconnecting on long running requests?

If I'm reading the docs & tracing the `net` dial functionality right, `net.Dialer.Timeout` deals more with a timeout for establishing the original TCP 3-way handshake.

I don't think that or the TCP KeepAlive interval is a knob we want to make configurable.  But it looks like this might be the option to accomplish your goals?

```
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
```

Reply:
I wonder if this needs to be mixed with the context deadline approach you mentioned above in option (2)?  Or if leaving that unset is perfectly fine because under the hood, these were getting cancelled by the TCP Transport layer under the hood purging idle connections of long running requests?